### PR TITLE
Helm chart prep

### PR DESCRIPTION
**Pull request by mateuszdrab**
- Add namespace support
- remove AWS hardcoded image names, deployment commands
- allow specifying IMAGE_NAME NAMESPACE when running make

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,23 +23,21 @@ deps:
 envars-webhook: $(shell find . -name '*.go')
 	TMPDIR=/var/tmp CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w" -o $@ ./cmd/envars-webhook
 
-image: envars-webhook deps
```
Comment (danfromtitan) [inline Makefile:27]: deps was added here to ensure module dependencies are resolved in a new environment (i.e. such as a build pipeline)

File: Makefile [hunk 2]
```diff
@@ -24,23 +23,21 @@ deps:
...
-	docker rmi $(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com/$(IMAGE_NAME):latest || true
```
Comment (danfromtitan) [inline Makefile:28]: The docker image needs the fully qualified name to push successfully to remote repository.

File: Makefile [hunk 3]
```diff
@@ -24,23 +23,21 @@ deps:
...
-	docker build --no-cache -t $(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com/$(IMAGE_NAME):latest .
+image: envars-webhook
+	docker rmi ${IMAGE_NAME} || true
+	docker build --no-cache -t ${IMAGE_NAME} .
 
 push:
-	aws ecr get-login-password --region $(AWS_REGION) | docker login --username AWS $(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com --password-stdin
-	aws ecr batch-delete-image --repository-name $(IMAGE_NAME) --image-ids imageDigest=$$(aws ecr list-images --repository-name $(IMAGE_NAME) | jq -r ' .imageIds[] | [ .imageDigest ] | @tsv ')
```
Comment (danfromtitan) [inline Makefile:33]: The lack of cleanup for the previous latest version would trigger a failure when u try to upload the image.

File: Makefile [hunk 4]
```diff
@@ -12,9 +12,8 @@
 
 # Makefile for building the Admission Controller webhook demo server and docker image.
 
-AWS_ACCOUNT_ID = $$(aws sts get-caller-identity --query Account --output text)
```
Comment (danfromtitan) [inline Makefile:15]: The docker image needs the fully qualified name to push successfully to remote repository.

File: Makefile [hunk 5]
```diff
@@ -24,23 +23,21 @@ deps:
...
-	aws ecr get-login-password --region $(AWS_REGION) | docker login --username AWS $(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com --password-stdin
-	aws ecr batch-delete-image --repository-name $(IMAGE_NAME) --image-ids imageDigest=$$(aws ecr list-images --repository-name $(IMAGE_NAME) | jq -r ' .imageIds[] | [ .imageDigest ] | @tsv ')
-	docker push $(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com/$(IMAGE_NAME):latest
+	docker push ${IMAGE_NAME}
 
 tls:
-	deploy/tls.sh
+	NAMESPACE=${NAMESPACE} deploy/tls.sh 
 
-deploy: undeploy
```
Comment (danfromtitan) [inline Makefile:39]: undeploy here would save you from having to run a "make undeploy"

File: Makefile [hunk 6]
```diff
@@ -24,23 +23,21 @@ deps:
...
-	deploy/deploy.sh create
```
Comment (danfromtitan) [inline Makefile:40]: The kubernetes manifest in deployment.yaml.template is declarative, thus apply should be used (i.e. create is for imperative constructions). Basically without apply the objects are not updated next time you deploy.

**Comment by danfromtitan** [comment]
There are numerous issues with this PR (see comments inline) which would break the current process. The process is driven from a makefile and it comes with image build & deployment all-in-one. I agree deployment should be separated in a helm chart, I didn't have time to look into that yet.

Bottom line is the only change from this PR that would make sense is turning NAMESPACE in a parameter, anything else would break the way things work.

**Comment by mateuszdrab** [comment]
Thanks for reviewing @danfromtitan 
I agree, I broke a few things - let me fix it up.
I'll comment on each review bits separately.

Perhaps the AWS specific commands should get their place in a dedicated part of the Makefile
For example, I use Azure ACR.

If it was a helm chart, the chart would reference a pre-built image in a registry where as building would happen in the pipeline which is why I added IMAGE_NAME. It can be set during the pipeline build/helm chart deployment

File: Makefile (hunk 4 above)
Comment (mateuszdrab) [reply Makefile:15]: I've created a IMAGE_NAME variable to handle providing custom image name which can have the repository provided

**Comment by danfromtitan** [comment]
@mateuszdrab , would you please explain the use case for [your last commit](https://github.com/danfromtitan/envars-from-node-labels/pull/5/commits/10e5a37dcc3433caf197a5fa1e69b2f3375ecbe9) ?

**Comment by mateuszdrab** [comment]
> @mateuszdrab , would you please explain the use case for [your last commit](https://github.com/danfromtitan/envars-from-node-labels/pull/5/commits/10e5a37dcc3433caf197a5fa1e69b2f3375ecbe9) ?

Damn, I meant to commit it to a different branch. Either way, I had a use case where I had to write out the value of the region node label to a file to be included in a dns zone file by bind9 and I achieved that through an init container but the feature wasn't there so I added it.

**Comment by danfromtitan** [comment]
@mateuszdrab , I captured the gist of your changes (i.e. namespace parametrization and init container support) in #7, since this one has a number od changes that are unrelated. Will close this PR and I will add helm support in a different commit (I have that WIP).

**Pull request closed without merging**